postprocessing/comstockpostproc/plotting_mixin.py

```diff
@@ -2110,3 +2113,629 @@ def plot_load_duration_curve(self, df, region, building_type, color_map, output_
+                                                                get_query_only=False
+                                                                )
+
+            # upgrade load data - all enduses
+            upgrade_ts_agg = run_data.agg.aggregate_timeseries(
+                                                                upgrade_id=upgrade_num.astype(str),
+                                                                enduses=(list(self.END_USES_TIMESERIES_DICT.values())+["total_site_electricity_kwh"]),
+                                                                restrict=[(('build_existing_model.building_type', [self.BLDG_TYPE_TO_SNAKE_CASE[btype]])),
+                                                                          ('state_abbreviation', [f"{state}"])
+                                                                          ],
+                                                                timestamp_grouping_func='hour',
+                                                                get_query_only=False
+                                                                )
+
+            # apply weights by building type
+            def apply_wgts(df):
+                # Identify columns that contain 'kwh' in their names
+                kwh_columns = [col for col in df.columns if 'kwh' in col]
+
+                # Apply the weight and add the suffix 'weighted'
+                weighted_df = df[kwh_columns].apply(lambda x: x * btype_wgt).rename(columns=lambda x: x + '_weighted')
+                # Concatenate the new weighted columns with the original DataFrame without the unweighted 'kwh' columns
+                df_wgt = pd.concat([df.drop(columns=kwh_columns), weighted_df], axis=1)
+
+                return df_wgt
+
+            # add baseline data
+            df_base_ts_agg_weighted = apply_wgts(df_base_ts_agg)
+            dfs_base.append(df_base_ts_agg_weighted)
+
+            # add upgrade data
+            df_upgrade_ts_agg_weighted = apply_wgts(upgrade_ts_agg)
+            dfs_up.append(df_upgrade_ts_agg_weighted)
+
+
+        # concatinate and combine baseline data
+        dfs_base_combined = pd.concat(dfs_base, join='outer', ignore_index=True)
+        dfs_base_combined = dfs_base_combined.groupby(['time'], as_index=False)[dfs_base_combined.loc[:, dfs_base_combined.columns.str.contains('_kwh')].columns].sum()
+        dfs_base_combined[self.UPGRADE_NAME] = 'baseline'
+
+        # concatinate and combine upgrade data
+        dfs_upgrade_combined = pd.concat(dfs_up, join='outer', ignore_index=True)
+        dfs_upgrade_combined = dfs_upgrade_combined.groupby(['time'], as_index=False)[dfs_upgrade_combined.loc[:, dfs_upgrade_combined.columns.str.contains('_kwh')].columns].sum()
+        dfs_upgrade_combined[self.UPGRADE_NAME] = upgrade_name
+
+        return dfs_base_combined, dfs_upgrade_combined
+
+    # plot
+    order_list = [
```

Comment:
There are 3 copies of this enduse list and corresponding color map in this file; is there a way to make this a Constant  that returns these things for reuse?